posthog/heatmaps/heatmaps_api.py

```diff
@@ -0,0 +1,220 @@
+
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.auth import TemporaryTokenAuthentication
+from posthog.hogql import ast
+from posthog.hogql.ast import Constant
+from posthog.hogql.base import Expr
+from posthog.hogql.parser import parse_expr, parse_select
+from posthog.hogql.query import execute_hogql_query
+from posthog.rate_limit import ClickHouseSustainedRateThrottle, ClickHouseBurstRateThrottle
+from posthog.schema import HogQLQueryResponse
+from posthog.utils import relative_date_parse_with_delta_mapping
+
+DEFAULT_QUERY = """
+            select pointer_target_fixed, relative_client_x, client_y, {aggregation_count}
+            from (
+                     select
+                        distinct_id,
+                        pointer_target_fixed,
+                        round((x / viewport_width), 2) as relative_client_x,
+                        y * scale_factor as client_y
+                     from heatmaps
+                     where {predicates}
+                )
+            group by `pointer_target_fixed`, relative_client_x, client_y
+            -- hogql enforces a limit (and only allows a max of 10000) but we don't really want one
+            -- see https://github.com/PostHog/posthog/blob/715a8b924e7c5dca7ae986bfdba6072b3999dbed/posthog/hogql/constants.py#L33
+            limit 10000000
+            """
+
+SCROLL_DEPTH_QUERY = """
+SELECT
+    bucket,
+    cnt as bucket_count,
+    sum(cnt) OVER (ORDER BY bucket DESC) AS cumulative_count
+FROM (
+    SELECT
+        intDiv(scroll_y, 100) * 100 as bucket,
+        {aggregation_count} as cnt
+    FROM (
+        SELECT
+           distinct_id, (y + viewport_height) * scale_factor as scroll_y
+        FROM heatmaps
+        WHERE {predicates}
+    )
+    GROUP BY bucket
+)
+ORDER BY bucket
+-- hogql enforces a limit (and only allows a max of 10000) but we don't really want one
+-- see https://github.com/PostHog/posthog/blob/715a8b924e7c5dca7ae986bfdba6072b3999dbed/posthog/hogql/constants.py#L33
+limit 10000000
+"""
+
+
+class HeatmapResponseItemSerializer(serializers.Serializer):
+    count = serializers.IntegerField(required=True)
+    pointer_y = serializers.IntegerField(required=True)
+    pointer_relative_x = serializers.FloatField(required=True)
+    pointer_target_fixed = serializers.BooleanField(required=True)
+
+
+def default_start_date():
+    return (datetime.now() - timedelta(days=7)).date()
+
+
+class HeatmapsRequestSerializer(serializers.Serializer):
+    viewport_width_min = serializers.IntegerField(required=False)
+    viewport_width_max = serializers.IntegerField(required=False)
+    type = serializers.CharField(required=False, default="click")
+    date_from = serializers.CharField(required=False, default="-7d")
+    date_to = serializers.DateField(required=False)
+    url_exact = serializers.CharField(required=False)
+    url_pattern = serializers.CharField(required=False)
```

Comment:
lol - only just noticed this value is not the same in the frontend. Explains why the url wasn't affecting anything...